server/api/users.js

```diff
@@ -0,0 +1,225 @@
+const router = require("express").Router();
+
+const { User, Order } = require('../db/index.js');
+
+//Finds, counts and serves all users
+router.get('/', (req, res, next) => {
+	User.findAll()
+	.then(users => res.send(users))
+	.catch(e => {
+		res.status(404);
+		next(e);
+	})
+})
+
+//Finds and serves a single user based on a primary key.
+router.get('/:userId', (req, res, next) => {
+	User.findByPk(req.params.userId)
+	.then(user => res.send(user))
+	.catch(e => {
+		res.status(404);
+		next(e);
+	})
+})
+
+//Creates a new user/signs a user up
+//Sets falsy fields in req.body that are allowed to be null to null
+router.post('/', (req, res, next) => {
+	const {
+		firstName,
+		lastName,
+		email,
+		password,
+		userType,
+		phone,
+		shippingAddress,
+		shippingCity,
+		shippingState,
+		shippingZip,
+		billingAddress,
+		billingCity,
+		billingState,
+		billingZip,
+	} = req.body
+
+	User.create({
+		firstName,
+		lastName,
+		email,
+		password,
+		userType,
+		phone: phone || null,
+		shippingAddress: shippingAddress || null,
+		shippingCity: shippingCity || null,
+		shippingState: shippingState || null,
+		shippingZip: shippingZip || null,
+		billingAddress: billingAddress || null,
+		billingCity: billingCity || null,
+		billingState: billingState || null,
+		billingZip: billingZip || null,
```

Comment:
When a new User is added to the database should there also be a 'key(loggedIn)' default to 'false'

Reply:
If the database model gives loggedIn a default value of false this shouldn't be a problem.